.github/workflows/action-build.yml

```diff
@@ -35,12 +50,51 @@ jobs:
       - name: Setup cross-rs
         if: runner.os == 'Linux'
         run: ./scripts/setup-cross.sh ${{ inputs.target }}
+      - name: Setup cosign for signing
+        if: ${{ inputs.runtime != 'common' && inputs.sign }}
+        uses: sigstore/cosign-installer@v3.3.0
+        with:
+          cosign-release: 'v2.2.2'
+      - name: Setup build profile
+        shell: bash
+        run: echo "OPT_PROFILE=release" >> ${GITHUB_ENV}
       - name: Build
         run: make build-${{ inputs.runtime }}
       - name: Run tests
         timeout-minutes: 5
         run: |
           make test-${{ inputs.runtime }}
+        if: ${{ inputs.arch == 'x86_64' }}
+      - name: Sign the binary
```

Comment:
Maybe we can move signing to a separate action that we call from here?
We can move the `Setup cosign for signing` part in there as well.

Reply:
Done